### PR TITLE
Update curationsortingextractor.py, merge_units

### PR DESCRIPTION
**Pull request by llobetv**
Hi,

I implemented idx method for features in to_merge function,
Is the method I used is ok for you? 

Cheer,
Victor

### CODE REVIEW

**Comment by alejoe91** [comment]
Hi @llobetv 

Can you also provide here a minimal example to test it? like the removal of duplicate spikes :)

I'll review the code now

File: spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py [hunk 1]
```diff
@@ -127,9 +130,12 @@ def merge_units(self, unit_ids):
             for feature_names in all_feature_names[1:]:
                 shared_feature_names.intersection_update(feature_names)
             shared_feature_names = list(shared_feature_names)
+        
             shared_features = []
+            shared_features_indx = []
```
Comment (alejoe91) [inline spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py:135]: can you call it `idxs` for consistency?

File: spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py [hunk 2]
```diff
@@ -140,21 +146,47 @@ def merge_units(self, unit_ids):
                 new_root.add_child(self._roots[root_index])
                 all_spike_trains.append(self._roots[root_index].get_spike_train())
                 for i, feature_name in enumerate(shared_feature_names):
-                    features = self.get_unit_spike_features(unit_id, feature_name)
-                    shared_features[i].append(features)
+                    if not feature_name[-5:] == "_idxs":
```
Comment (alejoe91) [inline spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py:149]: `feature_name.endswith('_idxs')`

Comment (alejoe91) [reply spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py:149]: alao at line169

**Comment by alejoe91** [comment]
Hi @llobetv 

Can you also provide here a minimal example to test it? like the removal of duplicate spikes :)

I'll review the code now

**Comment by llobetv** [comment]
Hi @alejoe91 
I make the change, it's actually more logical.

For the code I use it with some code for automatic merge :p

Here some code to test is:

```
sorting_path = ""
recording_path = ""
sorting = se.PhySortingExtractor(sorting_path)
recording = se.PhyRecordingExtractor(recording_path)
# This line exctract subpart of waveforms, which then will be save with idxs in sorting obj
waveforms = \
    st.postprocessing.get_unit_waveforms(recording, sorting,
                                         ms_before=1e3,
                                         ms_after=1e3, 
                                         max_channels_per_waveforms=1, 
                                         max_spikes_per_unit=10,
                                         memmap=True,
                                         n_jobs=1,
                                         verbose=False,
                                         save_property_or_features=True,
                                         recompute_info=True)

child_sorting = st.validation.quality_metric_classes.utils.curationsortingextractor.CurationSortingExtractor(sorting)
to_merge_list = [1, 2] # Choose 2 units that exist
child_sorting.merge_units(to_merge_list)  
```

File: spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py (hunk 1 above)
Comment (llobetv) [reply spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py]: Done

File: spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py (hunk 2 above)
Comment (llobetv) [reply spiketoolkit/validation/quality_metric_classes/utils/curationsortingextractor.py]: Done

**Comment by alejoe91** [comment]
@llobetv in the code snippet why are you setting ms_before and after to 1e3? that's in ms!

I think that it looks good :)

**Comment by llobetv** [comment]
> @llobetv in the code snippet why are you setting ms_before and after to 1e3? that's in ms!
I use second in my function, I forgot to remove them 1e3 ^^.

**Comment by alejoe91** [comment]
Tested it and it works perfectly

**Comment by alejoe91** [comment]
Easier to use it with this https://github.com/SpikeInterface/spiketoolkit/pull/353
